### ISSUE TITLE
API string search looks for all words separated by symbols instead of entire string

### ISSUE THREAD

**Issue by BrickmanRodrigo** [dotCMS/core#19371]
**Describe the bug**
As the title said, I'm trying to search a string like dotcms-itemA like this:

https://cmsurl.something/api/content/render/false/depth/3/query/+contentType:MyContentType%20+MyContentType.fieldName:dotcms-itemA

But instead of returning only the item that matches the string exactly, it returns all items in that content type with the field "fieldName" containing dotcms OR itemA. Also, the order of the words around the "-" doesn't matter, the issue happens anyways.

The weirdest thing is that this happens in our production environment, but it doesn't happen in the test environment, even if both have the exact same dotCMS version.

Also, in the production environment adding the "_dotraw" at the end of the field name like this {...}/+contentType:MyContentType%20+MyContentType.fieldName_dotraw:dotcms-itemA works, but in the test environment it doesn't

dotCMS Version: 5.2.3

**To Reproduce**
I don't really know how to reproduce this in another environment.

**Expected behavior**
It should return the content with the string exactly matching the query.

Thanks!

**Comment by john-thomas-dotcms** [comment]
@BrickmanRodrigo the default behavior you're seeing is actually how it's expected to work.  When your content is indexed, Elasticsearch automatically tokenizes it - breaking the value of the field into words, as separated by white space and punctuation.  Then, when you search on the field normally, it's searching against those tokens, not against the entire value of the field.  Elasticsearch also automatically tokenizes the search string (e.g. `dotcms-itemA`).

The way to do what you want to do is, as you tried, to search against the `_dotraw` version of the field instead.  But note that when searching for a longer string, you must enclose the string you're searching for in quotes to tell ES not to tokenize that (so `+MyContentType.fieldName_dotraw:"dotcms-itemA"`).

So bottom line, it appears that your production environment is behaving exactly as it's designed (and intended) to.  If you're seeing different results in your test and production environments - with the exact same content type and content items in both - then that is a different issue, and most likely related to the configuration of the test environment (which appears to be having the unexpected behavior).
